packages/maskbook/src/web3/hooks/useEtherTransferCallback.ts

```diff
@@ -67,10 +70,18 @@ export function useEtherTransferCallback(amount?: string, recipient?: string, me
 
         // step 1: estimate gas
         const estimatedGas = await Services.Ethereum.estimateGas(config, chainId)
+        const gas = addGasMargin(estimatedGas, 2000)
```

Comment:
It's to late to adjust amount in this hook. User will be noticed the wrong amount after finishing the transaction. Better alert it when the user clicks the `MAX` button on the transfer dialog.